conans/model/build_info.py

```diff
@@ -126,13 +257,89 @@ def _get_cpp_info():
             result.libdirs.append(DEFAULT_LIB)
             result.bindirs.append(DEFAULT_BIN)
             result.resdirs.append(DEFAULT_RES)
-            result.builddirs.append("")
+            result.builddirs.append(DEFAULT_BUILD)
             return result
 
         return self.configs.setdefault(config, _get_cpp_info())
 
 
+class Component(object):
+
+    def __init__(self, name, root_folder):
+        self._rootpath = root_folder
+        self.name = name
+        self.deps = []
```

Comment:
There is no need for that, you can perform the checks in the `_sorted_components` that will "compute" the graph. It can error when loops or undefined nodes.

Reply:
Done in `_sorted_components` as @lasote indicated. I think is more flexible and less complex to implement